Source/gendung.cpp

```diff
@@ -477,24 +474,26 @@ void DRLG_InitTrans()
 	TransVal = 1;
 }
 
-void DRLG_RectTrans(int x1, int y1, int x2, int y2)
+void DRLG_RectTrans(Rectangle area)
 {
-	for (int j = y1; j <= y2; j++) {
-		for (int i = x1; i <= x2; i++) {
+	Point position = area.position;
+	Size size = area.size;
+
+	for (int j = position.y; j <= position.y + size.height; j++) {
+		for (int i = position.x; i <= position.x + size.width; i++) {

```

Comment:
Yeah there are a few more cases of this. It's something I would like to address in a later PR when in full call path for these regions have been cleaned up to use Point and Rectangle so that it is easier to see where things are being applied and have logic issues or off by one errors.

Reply:
Need to convert both extent and origin to world space. I ended up getting nerd-sniped and got this:
https://github.com/diasurgical/devilutionX/commit/125da655d587344c311e7997c9e4bb395dff8ea8

(no PR because despite declaring the constructor explicit it's still considered a candidate for an implicit rectangle declaration? previous commit shows the fun getting this to compile.)

Reply:
Lol, that feeling when you just want something to be done :D

```cpp
void DRLG_MRectTrans(Point origin, Point extent)
{
	DRLG_MRectTrans({ origin, { extent.x - origin.x, extent.y - origin.y } });
}
```

Feel free to improve further then merged :joy: :white_flag: